docs/build.md

```diff
@@ -15,3 +15,74 @@ To run the CMake targets:
 2. `cmake ..` generates project configuration in the build directory
 3. `cmake --build .` to build the default target,
     or `cmake --build . --target <target_name>` to specify a target to run from above
+
+
+## Running it locally
+This section outlines the environment variables and files necessary to successfully compile and run the Go binary, then connect it to an [Azure Kubernetes Service](https://docs.microsoft.com/en-us/azure/aks/intro-kubernetes).
+
+### Obtain Azure Credentials
+
+In order to run the Go binary locally and control a remote AKS server, you need Azure credentials. These will be stored in a JSON file in your home directory.
+
+Follow [these instructions](https://docs.microsoft.com/en-us/dotnet/api/overview/azure/containerinstance?view=azure-dotnet#authentication) to create the `$HOME/.azure/azureAuth.json` file. The file is generated via:
+```bash
+az ad sp create-for-rbac --subscription <your-azure-subscription-id> --sdk-auth > $HOME/.azure/azureAuth.json
+```
+The file will contain a JSON blob with the following shape:
+```json
+{
+  "clientId": "...",
+  "clientSecret": "...",
+  "subscriptionId": "<your-azure-resource-group>",
+  "tenantId": "...",
+  "activeDirectoryEndpointUrl": "https://login.microsoftonline.com",
+  "resourceManagerEndpointUrl": "https://management.azure.com/",
+  "activeDirectoryGraphResourceId": "https://graph.windows.net/",
+  "sqlManagementEndpointUrl": "https://management.core.windows.net:8443/",
+  "galleryEndpointUrl": "https://gallery.azure.com/",
+  "managementEndpointUrl": "https://management.core.windows.net/"
+}
+```
+
+### Startup Script
+1. Create a `.dev` directory in your repo. This is already in `.gitignore` and will not be comitted.
+1. Create an executable bash file in the root directory of this repo: `touch .dev/start.sh && chmod +x .dev/start.sh`
+1. Add the following bash script to the `start.sh` file:
+```bash
+#!/bin/bash
+
+set -auexo pipefail
+
+export AZURE_AUTH_LOCATION=$HOME/.azure/azureAuth.json
+export KUBERNETES_WATCHNAMESPACE=default
```

Comment:
shift `KUBERNETES_WATCHNAMESPACE` this after the `APPGW_*` env variables and add a new line after `AZURE_AUTH_LOCATION`,.